### PR TITLE
9373: Ability to be able to enter comma separated latitude/longitude …

### PR DESCRIPTION
**Pull request by SNoiraud**
…pairs

### CODE REVIEW

**Comment by Nick-Hall** [comment]
I think that three fields for latitude and longitude could be confusing.  Perhaps displaying either one ot two based on a configuration option would be better.

The place reference editor should also be changed for consistency.

**Comment by SNoiraud** [comment]
I don't agree with your view. I think the three fields must be present.
I think we must arrange them and comment for the user to choose either the two standard fields, or the new one.
We always need to have the two first fields, because they show if the entered fields are conform to the latitude or longitude format. If you use google or OSM, it will be good (for the moment), but I don't know what happens with other providers that some users can use.

**Comment by SNoiraud** [comment]
No more comments ?
I apply this to EditPlaceRef.

**Comment by sam-m888** [comment]
Tested and works for me :+1: 

 - [ ] Update related wikipages if accepted.

**place-reference-editor**

place-reference-editor-g5

**edit-place dialog**
edit-place-g5

**Comment by eno93** [comment]
A bit wordy and cluttered but worked for me.

**Comment by Nick-Hall** [comment]
This doesn't work for me.  If I paste "1,2" into the new entry field the latitude and longitude fields are updated, but when I click "OK" the values are not saved.

I still don't like the user interface.  It would be better to have a label "Coordinates:" followed by a new widget.  This widget could display either one or two fields depending on a user preference.

File: gramps/gui/editors/editplace.py [hunk 1]
```diff
@@ -176,6 +176,24 @@ def _setup_fields(self):
...
+            self.db.readonly)
+
+    def set_latlongitude(self, value):
+        try:
+            coma = value.index(',')
+            self.longitude.set_text(value[coma+1:])
+            self.latitude.set_text(value[:coma])
+            self.top.get_object("lat_entry").validate(force=True)
+            self.top.get_object("lon_entry").validate(force=True)
+        except:
```
Comment (Nick-Hall) [inline gramps/gui/editors/editplace.py]: It is good practice to catch specific exceptions.

Comment (SNoiraud) [inline gramps/gui/editors/editplace.py]: I didn't want to reinvent the wheel. So I use what already exists.
You enter 1,2. Neither google, openstreetmap or other map services will give you these values.
It is written : **copy/paste**. You entered manually these values.
If you want to create a new widget, I don't like this : There are too many widgets in gramps. I think this is clear like that. I can add some controls if you want.

**Comment by Nick-Hall** [comment]
I still can't get this to work.  As an example, I tried to paste the coordinates of Buckingham Palace from Google maps.  It gave "51.500838, -0.142281". When pasted, the values were correctly split into the latitude and longitude fields, but were not saved when I clicked "OK".

**Comment by SNoiraud** [comment]
I saw the problem. The value were not saved.
Found a bug in gramps/gui/widgets/monitoredwidgets.py : get_value needed an argument!

**Comment by SNoiraud** [comment]
Tested with :  
1,2 => OK
51° 30′ 04″ N, 0° 08′ 31″ W => OK
51.500838, -0.142281 => OK
The values are saved.

**Comment by romjerome** [comment]
I am thinking on some ideas about place, set of coordinates, batch and separators.

Two local branches have been generated:
* [Place separators](https://github.com/romjerome/gramps_cryptic_rebase/tree/place_separators) see bug reports _7276_ and _9851_
* [Reverse geocoding](https://github.com/romjerome/gramps_cryptic_rebase/tree/reverse_geocoding), which is based on Serge's patch (_bug report 5742 and 9595_)

We could go further: to have one separator character, set by user (_config preference_), used for importing a set of values, used for batch on place title, gedcom handling or whatever need for a separator. Yes, close to csv import but technical stuff will be limited.

**Comment by romjerome** [comment]
@SNoiraud 

Did you see this "[meet-up](http://www.geneanet.org/blog/post/2016/12/genealogie-lab-1-cartographie-et-geolocalisation-en-genealogie)"?

**Comment by kulath** [comment]
I have only looked at sam's screenshots above. I like the three fields (even though I agree with Enno on it being a bit cluttered and wordy) it is very clear what to expect and easy to input the data in whatever way is convenient.

I would have put a new line after the words"following field", so the entry box is under latitude, but that's just a personal preference.

**Comment by SNoiraud** [comment]
I accidently merged my FR9373 branch with master. I had two modifications about the two glade files above.
I pushed it and now my two files are correct, but there are no trace that I made modifications.
What can I do ? recreate a branch and apply my patch ?

**Comment by Nick-Hall** [comment]
@SNoiraud Fix your local branch, then force push to your public fork.

**Comment by SNoiraud** [comment]
Unable to do that. I think my repository is unusable now.
I tried several solution, but nothing works.
Is there a way to suppress my fork and to recreate it ? I can't do that with the web.

**Comment by Nick-Hall** [comment]
Rebased.

**Comment by SNoiraud** [comment]
Thanks for this branch. 

How to do now with the master and others branches now ?
If I make enter "git checkout master", I get : 
gramps/gen/db/generic.py: needs merge
gramps/gen/lib/test/serialize_test.py: needs merge
gramps/gen/merge/diff.py: needs merge
gramps/gui/editors/edittaglist.py: needs merge
gramps/gui/managedwindow.py: needs merge
gramps/gui/plug/_windows.py: needs merge
gramps/gui/undohistory.py: needs merge
gramps/gui/views/tags.py: needs merge
gramps/gui/widgets/progressdialog.py: needs merge
gramps/plugins/db/dbapi/dbapi.py: needs merge
gramps/plugins/test/test_imports.py: needs merge
gramps/plugins/tool/dumpgenderstats.py: needs merge
po/cs.po: needs merge
po/fi.po: needs merge
po/fr.po: needs merge
po/gramps.pot: needs merge
po/nl.po: needs merge
po/ru.po: needs merge
error: you need to resolve your current index first

**Comment by Nick-Hall** [comment]
@SNoiraud It looks like you are in the middle of a merge and still have conflicts that need to be resolved.  Do you have anything in master that you want to keep?  If not, you could abort the merge with `git reset --merge` and then remove any unwanted commits with `git reset --hard <commit>`.  Be careful when using `git reset`.

When you have fixed your local master, you can force push it to your fork.

**Comment by SNoiraud** [comment]
I think I solve my problem. many thanks for your help.

**Comment by sam-m888** [comment]
Revised with larger field to paste into.

**place-reference-editor**

place-reference-editor-revised


**edit-place dialog**

edit-place dialog-revised

**Comment by sam-m888** [comment]
Thank you
